### ISSUE TITLE
Support for -Z asm-comments

### ISSUE THREAD

**Issue by GabrielMajeri** [gnzlbg/cargo-asm#58]
A [recent rust-lang PR](https://github.com/rust-lang/rust/pull/53290) added support for LLVM assembly comment output. While only available on nightly, it would be great if `cargo asm` could support them.

Currently, the following commands don't work:

```sh
$ RUSTFLAGS="-Z asm-comments" cargo asm
thread 'main' panicked at 'line starts with _ but we failed to parse the label: ...
```

**Comment by gnzlbg** [comment]
Yep, I've asked if there are any plans to stabilize that flag but there aren't any already. We can detect if the compiler is a nightly compiler, and pass it the flag if that's the case.
